src/EditorFeatures/Test2/Utilities/TestNotificationService.vb

```diff
@@ -0,0 +1,33 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports Microsoft.CodeAnalysis.Notification
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Utilities
+    Friend Class TestNotificationService
+        Implements INotificationService
```

Comment:
Rather than implement this, you can use the real `EditorNotificationService` and cast to the internal `INotificationServiceCallback` interface which lets you set an `Action<>` `NotificationCallback` to be invoked instead of the real notification getting fired.  `INotificationServiceCallback` was created specifically for testing.